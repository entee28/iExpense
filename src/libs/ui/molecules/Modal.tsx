import React, { FC } from 'react'
import { StyleSheet, ViewStyle } from 'react-native'
import RNModal, { ModalProps as RNModalProps } from 'react-native-modal'
import { useSafeAreaInsets } from 'react-native-safe-area-context'
import { SCREEN_PADDING_HORIZONTAL } from '../constants'

export type ModalProps = {
  containerStyle?: ViewStyle
} & Partial<RNModalProps>

export const Modal: FC<ModalProps> = ({
  containerStyle,
  children,
  ...props
}) => {
  const { top, bottom } = useSafeAreaInsets()

  return (
    <RNModal
      {...props}
      backdropOpacity={0.5}
      backdropTransitionOutTiming={0}
      style={[
        styles.modal,
        {
          paddingTop: top + SCREEN_PADDING_HORIZONTAL,
          paddingBottom: bottom + SCREEN_PADDING_HORIZONTAL,
          paddingHorizontal: SCREEN_PADDING_HORIZONTAL
        },
        containerStyle
      ]}>
      {children}
    </RNModal>
  )
}

const styles = StyleSheet.create({
  modal: {
    margin: 0,
    justifyContent: 'center'
  }
})
