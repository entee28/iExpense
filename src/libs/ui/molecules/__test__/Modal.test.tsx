import { render } from '@testing-library/react-native'
import { Modal } from '../Modal'

it('should match snapshot', () => {
  const modal = render(<Modal isVisible={true} />).toJSON()

  expect(modal).toMatchSnapshot()
})
