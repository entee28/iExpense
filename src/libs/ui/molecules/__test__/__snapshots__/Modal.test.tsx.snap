// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<react-native-modal
  backdropOpacity={0.5}
  backdropTransitionOutTiming={0}
  isVisible={true}
  style={
    [
      {
        "justifyContent": "center",
        "margin": 0,
      },
      {
        "paddingBottom": 24,
        "paddingHorizontal": 24,
        "paddingTop": 24,
      },
      undefined,
    ]
  }
/>
`;
